Refactor testdriver-vendor.js to make better use of async/await and promises

marcoscaceres opened the pull request:
#### 5f95c4f4489da355e5c59a0d7f7877c8f081222d
<pre>
Refactor testdriver-vendor.js to make better use of async/await and promises
<a href="https://bugs.webkit.org/show_bug.cgi?id=248248">https://bugs.webkit.org/show_bug.cgi?id=248248</a>
rdar://102613798

Make testdriver-vendor.js a bit more ES idiomatic, plus fix some incorrect
error handling with promises.

Also provides slightly better error messages.

Reviewed by Tim Nguyen.

* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-custom-formats-write-read.tentative.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-html-script-removal.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-read-resource-load.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-read-sanitize.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-promise-write-blobs-read-blobs.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-svg-script-removal.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-unsanitized-html-formats-write-read.tentative.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-unsanitized-plaintext-formats-write-read.tentative.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-blobs-read-blobs.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-html-read-html.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-image-read-image.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-svg-read-svg.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/read-on-detaching-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/write-on-detaching-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/write-read-on-detached-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/writeText-readText-on-detached-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/feature-policy/clipboard-write/clipboard-write-disabled-by-feature-policy.tentative.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/feature-policy/clipboard-write/clipboard-write-enabled-by-feature-policy.tentative.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/feature-policy/clipboard-write/clipboard-write-enabled-on-self-origin-by-feature-policy.tentative.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/readText-denied.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/readText-granted.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/writeText-denied.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/writeText-granted.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-write-read.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-write-readText.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-writeText-read.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-writeText-readText.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-overscroll-behavior/overscroll-behavior-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/input/mouse-wheel-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/input/snap-area-overflow-boundary-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/selection-target-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/snap-at-user-scroll-end-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-transforms/scroll-preserve-3d-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/scrolling/iframe-chains-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fullscreen/api/element-request-fullscreen-options.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-label-element/proxy-modifier-click-to-associated-element.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-canceling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/inert-node-is-uneditable-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/compat/pointerevent_mouseevent_key_pressed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
(pause):
(async dispatchMouseActions):
(async dispatchTouchActions):
(async if):
(window.test_driver_internal.send_keys):
(window.test_driver_internal.click):
(async let):
(window.test_driver_internal.action_sequence):
(async switch):
(window.test_driver_internal.set_permission):
(window.test_driver_internal.delete_all_cookies): Deleted.
(window.test_driver_internal.generate_test_report): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/accesskey.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/order-of-events/focus-events/focus-management-expectations-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/order-of-events/mouse-events/wheel-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/order-of-events/mouse-events/wheel-scrolling-expected.txt:

Canonical link: <a href="https://commits.webkit.org/257030@main">https://commits.webkit.org/257030@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/866555715c33bca17ea770c7fcdcfab3d0b13c5b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97648 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6894 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30814 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/107145 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/167408 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/7258 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35655 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/90044 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/103800 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/103293 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/5434 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/84279 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/32445 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/87314 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/89104 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/75365 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/text/state-changed (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/892 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20551 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/882 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/22034 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4839 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5699 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/44496 "Found 1 new test failure: fast/events/blur-remove-parent-crash.html (failure)") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/2132 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/41426 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -263,16 +259,16 @@ window.test_driver_internal.action_sequence = function(sources)
             break;
         default:
-            return Promise.reject(new Error(`Unknown source type "${action.type}".`));
+            throw new Error(`Unknown source type "${action.type}".`);
```
since you're touching this, this should actually be `source.type` (many tests will need a rebaseline, but I think it's worth doing the change).

marcoscaceres replied:
ok, cool. Let's see what comes back..

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -289,45 +285,46 @@ window.test_driver_internal.action_sequence = function(sources)
-        return dispatchMouseActions(pointerSource.actions, pointerType);
+        await dispatchTouchActions(pointerSource.actions, { insertPauseAfterPointerUp: true });
+    if (["mouse", "pen"].includes(pointerType))
+        await dispatchMouseActions(pointerSource.actions, pointerType);
```
This is a behavior difference fwiw, `return` stops the function execution, and prevents from going to the next `if` branch. `await` doesn't.

marcoscaceres replied:
Oh you are right. Oops.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -9,18 +9,16 @@ function logDebug(callback)
+    return new Promise((resolve, reject) => {
         setTimeout(async () => {
             eventSender.dragMode = false;
 
```
Can this be flattened further to make more full use of async/await?

e.g. `await new Promise(r => setTimeout(r));`

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -221,16 +220,15 @@ window.test_driver_internal.send_keys = function(element, keys)
+        throw new Error("window.eventSender is undefined.");
 
     if (testRunner.isIOSFamily && testRunner.isWebKit2) {
         return new Promise((resolve) => {
```
should be `await new Promise((resolve) => {...});` followed by `return;` to preserve old behavior.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -9,18 +9,16 @@ function logDebug(callback)
+    await new Promise((resolve, reject) => {
         setTimeout(async () => {
             eventSender.dragMode = false;
 
```
Can we flatten this now that we're using async await?

```
 // Wait for tick
await new Promise(resolve => setTimeout(resolve, 0));
eventSender.dragMode = false;
for (let action of actions) {
    ...
    throw new Error(`Unknown action type "${action.type}".`));
...
```

marcoscaceres replied:
yeah, much nicer!

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -285,9 +280,7 @@ window.test_driver_internal.action_sequence = function(sources)
     logDebug(() => JSON.stringify(pointerSource));
-
     if (pointerType === "touch")
         return dispatchTouchActions(pointerSource.actions);
```
should be `await` then `else if` to preserve old behavior, given both methods return a Promise.

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
```diff
@@ -9,59 +9,54 @@ function logDebug(callback)
-                default:
-                    return Promise.reject(new Error(`Unknown action type "${action.type}".`));
-                }
+    await setTimeout(new Promise(resolve => resolve()), 0);
```
Does this work as expected? I think it should be the other way round:

`await new Promise(resolve => setTimeout(resolve, 0));`

marcoscaceres replied:
oh whoops. Good catch!

marcoscaceres replied:
This is why I'd like to add more TypeScript annotations... and hopefully get us using a bit more TypeScript. It catches silly mistakes like this in the code editor.

marcoscaceres committed:
Refactor testdriver-vendor.js to make better use of async/await and promises
https://bugs.webkit.org/show_bug.cgi?id=248248
rdar://102613798

Make testdriver-vendor.js a bit more ES idiomatic, plus fix some incorrect
error handling with promises.

Also provides slightly better error messages.

Reviewed by Tim Nguyen.

* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-custom-formats-write-read.tentative.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-html-script-removal.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-read-resource-load.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-navigator-clipboard-read-sanitize.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-promise-write-blobs-read-blobs.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-svg-script-removal.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-unsanitized-html-formats-write-read.tentative.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-unsanitized-plaintext-formats-write-read.tentative.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-blobs-read-blobs.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-html-read-html.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-image-read-image.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/async-write-svg-read-svg.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/read-on-detaching-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/write-on-detaching-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/write-read-on-detached-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/detached-iframe/writeText-readText-on-detached-iframe.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/feature-policy/clipboard-write/clipboard-write-disabled-by-feature-policy.tentative.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/feature-policy/clipboard-write/clipboard-write-enabled-by-feature-policy.tentative.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/feature-policy/clipboard-write/clipboard-write-enabled-on-self-origin-by-feature-policy.tentative.https.sub-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/readText-denied.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/readText-granted.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/writeText-denied.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/permissions/writeText-granted.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-write-read.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-write-readText.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-writeText-read.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/clipboard-apis/text-write-read/async-writeText-readText.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-overscroll-behavior/overscroll-behavior-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/input/mouse-wheel-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/input/snap-area-overflow-boundary-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/selection-target-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scroll-snap/snap-at-user-scroll-end-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-transforms/scroll-preserve-3d-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-mousewheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/non-passive-wheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-mousewheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-body-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-div-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-document-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-root-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/non-cancelable-when-passive/passive-wheel-event-listener-on-window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/dom/events/scrolling/iframe-chains-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/fullscreen/api/element-request-fullscreen-options.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-button-element/active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/checkable-active-onblur-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/the-label-element/proxy-modifier-click-to-associated-element.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/dialog-canceling-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-dialog-element/inert-node-is-uneditable-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/compat/pointerevent_mouseevent_key_pressed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/pointerevents/pointerevent_touch-action-keyboard-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js:
(pause):
(async dispatchMouseActions):
(async dispatchTouchActions):
(async if):
(window.test_driver_internal.send_keys):
(window.test_driver_internal.click):
(async let):
(window.test_driver_internal.action_sequence):
(async switch):
(window.test_driver_internal.set_permission):
(window.test_driver_internal.delete_all_cookies): Deleted.
(window.test_driver_internal.generate_test_report): Deleted.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/accesskey.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/order-of-events/focus-events/focus-management-expectations-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/order-of-events/mouse-events/wheel-basic-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/uievents/order-of-events/mouse-events/wheel-scrolling-expected.txt:

Canonical link: https://commits.webkit.org/257030@main
